ecdsa_public_key.go

```diff
@@ -70,24 +73,67 @@ func _ECDSAPublicKeyFromBytesDer(byt []byte) (*_ECDSAPublicKey, error) {
 		return &_ECDSAPublicKey{}, errByteArrayNull
 	}
 
-	given := hex.EncodeToString(byt)
-	result := strings.ReplaceAll(given, _ECDSAPubKeyPrefix, "")
-	decoded, err := hex.DecodeString(result)
-	if err != nil {
-		return &_ECDSAPublicKey{}, err
+	type AlgorithmIdentifier struct {
+		Algorithm  asn1.ObjectIdentifier
+		Parameters asn1.ObjectIdentifier
 	}
 
-	if len(decoded) != 33 {
-		return &_ECDSAPublicKey{}, _NewErrBadKeyf("invalid public key length: %v bytes", len(byt))
+	type PublicKeyInfo struct {
+		AlgorithmIdentifier AlgorithmIdentifier
+		PublicKey           asn1.BitString
 	}
 
-	key, err := crypto.DecompressPubkey(decoded)
+	key := &PublicKeyInfo{}
+	_, err := asn1.Unmarshal(byt, key)
 	if err != nil {
-		return &_ECDSAPublicKey{}, err
+		return nil, err
+	}
+
+	// Check if the parsed key uses ECDSA public key algorithm
+	ecdsaPublicKeyAlgorithmOID := asn1.ObjectIdentifier{1, 2, 840, 10045, 2, 1}
+	if !key.AlgorithmIdentifier.Algorithm.Equal(ecdsaPublicKeyAlgorithmOID) {
+		return nil, errors.New("public key is not an ECDSA public key")
+	}
+
+	// Check if the parsed key uses secp256k1 curve
+	secp256k1OID := asn1.ObjectIdentifier{1, 3, 132, 0, 10}
```

Comment:
**thought:** ‏We can probably extract those `asn1.ObjectIdentifier` that we use at a number of different places, have them defined in a single place and just reuse them where needed.

Reply:
Not sure, its used in 2 places in this file, so not that much of duplication. Happy either way.